### PR TITLE
Update release-coredns

### PR DESCRIPTION
**Pull request by yongtang**
Update golang download link.

/cc @chrisohaver @miekg

### CODE REVIEW

File: release-coredns [hunk 1]
```diff
@@ -55,11 +55,11 @@ function latestGo {
     local VERSION="$2"
 
     if [[ -z "$VERSION" ]]; then
-        local LATEST=$(curl -s 'https://golang.org/dl/?mode=json' | jq -r '.[0].version')
-        local GOLANG=https://dl.google.com/go/${LATEST}.linux-amd64.tar.gz
+        local LATEST=$(curl -s 'https://go.dev/dl/?mode=json' | jq -r '.[0].version')
```
Comment (miekg) [inline release-coredns]: for my local script I simply added a `-L` to curl, even with this new url that might be prudent to do

Comment (miekg) [reply release-coredns]:
```suggestion
        local LATEST=$(curl -sL 'https://go.dev/dl/?mode=json' | jq -r '.[0].version')
```

**Comment by yongtang** [comment]
Thanks @miekg  The PR has been updated.

**Comment by miekg** [comment]
I should _really_ fix getting this deployed more automatically. We could potentially make the script check it's own version with one here on github and download a new one?

**Comment by chrisohaver** [comment]
Or perhaps modify the web hook script on drone that currently executes this script to download latest and execute.

**Pull request merged**